spec/controllers/coronavirus_form/product_details_controller_spec.rb

```diff
@@ -203,6 +203,99 @@
+        end
+
+        it "errors if the product quantity is below zero" do
+          post :submit, params: params.merge(product_quantity: "-10")
+          expect(response).to render_template(current_template)
+        end
+      end
+
+      context "product_cost" do
+        before do
+          params.merge!(equipment_type: "Gloves")
+        end
+
+        it "errors if the user doesn't provide a product_cost" do
+          post :submit, params: params.except(:product_cost)
+          expect(response).to render_template(current_template)
+        end
+
+        it "errors if the user enters an invalid product_cost" do
+          post :submit, params: params.merge(product_cost: "Ten pounds")
+          expect(response).to render_template(current_template)
+        end
+
+        it "errors if the product_cost is below zero" do
+          post :submit, params: params.merge(product_cost: "-10")
+          expect(response).to render_template(current_template)
+        end
+
+        it "errors if the product_cost has more than two decimal places" do
+          post :submit, params: params.merge(product_cost: "10.99999")
+          expect(response).to render_template(current_template)
+        end
+
+        it "allows the product_cost to be zero" do
+          post :submit, params: params.merge(product_cost: "0")
+          expect(response).to redirect_to(additional_product_path)
+        end
+
+        it "removes £ from product_cost" do
+          post :submit, params: params.merge(product_cost: "£10.99")
+          expect(session[:product_details].first[:product_cost]).to eq("10.99")
+        end
+
+        it "removes words from product_cost" do
+          post :submit, params: params.merge(product_cost: "10.99 GBP")
+          expect(session[:product_details].first[:product_cost]).to eq("10.99")
+        end
+      end
+
+      context "lead_time" do
+        before do
+          params.merge!(equipment_type: "Gloves")
+        end
+
+        it "errors if the user doesn't provide a lead_time" do
+          post :submit, params: params.except(:lead_time)
+          expect(response).to render_template(current_template)
+        end
+
+        it "errors if the user enters an invalid lead_time" do
+          post :submit, params: params.merge(lead_time: "Ten days")
+          expect(response).to render_template(current_template)
+        end
+
+        it "errors if the lead_time is below zero" do
+          post :submit, params: params.merge(lead_time: "-10")
+          expect(response).to render_template(current_template)
+        end
+
+        it "allows the lead_time to be zero" do
+          post :submit, params: params.merge(lead_time: "0")
+          expect(response).to redirect_to(additional_product_path)
+        end
+
+        it "removes words from lead_time" do
+          post :submit, params: params.merge(lead_time: "10 days")
+          expect(session[:product_details].first[:lead_time]).to eq("10")
```

Comment:
Hm, this might be confusing is someone writes "10 weeks" into the field. Do you think we should return an error instead?

Reply:
The requested validation was just to strip out "days". But then I thought, what if the user enters "1 day", so I stripped out all words. Now I'm not sure which approach to take.